pandas/core/indexes/api.py

```diff
@@ -89,6 +89,19 @@ def get_objs_combined_axis(
     Index
     """
     obs_idxes = [obj._get_axis(axis) for obj in objs]
+    if all_indexes_same(obs_idxes):
+        idx = obs_idxes[0]
+        if sort:
+            try:
+                idx = idx.sort_values()
+                copy = False
+            except TypeError:
+                pass
+        if copy:
```

Comment:
shouldn't this come before `sort`? Presumably if sort=True, copy=True we want to copy first then sort?

Reply:
Doesn't `sort_values` always return a copy?

Reply:
So it does

Maybe OT but should we then warn if copy=false and sort=true (or do we do that already?)

Reply:
I'd say no. The only cases where it's possible to not copy are: when all indices are references to the same object or (now) when all objects have the same value. Otherwise a new value is created, so why even copy?

This is also often not super user visible, since the only time this argument is even passed is in `concat`, where the documentation is clear `copy=False` is "best effort" instead of a strictly enforced rule.